ambari-server/src/main/java/org/apache/ambari/server/configuration/Configuration.java

```diff
@@ -2851,6 +2854,41 @@ private static Properties readConfigFile() {
     return properties;
   }
 
+  /**
+   * Removing the given properties from ambari.properties (i.e. at upgrade time)
+   *
+   * @param propertiesToBeCleared
+   *          the properties to be removed
+   * @throws AmbariException
+   *           if there was any issue when clearing ambari.properties
+   */
+  public void removePropertiesFromAmbariProperties(Collection<String> propertiesToBeRemoved) throws AmbariException {
+    try {
+      final File ambariPropertiesFile = new File(Configuration.class.getClassLoader().getResource(Configuration.CONFIG_FILE).getPath());
+      final List<String> ambariPropertiesLines = FileUtils.readLines(ambariPropertiesFile, Charset.defaultCharset());
+      final StringBuilder newAmbariProperties = new StringBuilder();
+      for (String line : ambariPropertiesLines) {
+        if (!lineNeedsToBeRemoved(line, propertiesToBeRemoved)) {
+          newAmbariProperties.append(line).append(System.getProperty("line.separator"));
+        }
+      }
+      FileUtils.deleteQuietly(ambariPropertiesFile);
+      FileUtils.writeStringToFile(ambariPropertiesFile, newAmbariProperties.toString(), Charset.defaultCharset());
+
+      // reloading properties
+      this.properties = readConfigFile();
+    } catch (IOException e) {
+      throw new AmbariException("Error while clearing ambari.properties", e);
+    }
+  }
+
+  private boolean lineNeedsToBeRemoved(String line, Collection<String> propertiesToBeRemoved) {
+    if (StringUtils.isNotBlank(line)) {
+      return propertiesToBeRemoved.stream().filter(propertyToBeCleared -> line.trim().indexOf(propertyToBeCleared) > -1).findFirst().isPresent();
```

Comment:
maybe anyMatch() would be shorter

Reply:
This implementation also removes properties with prefix and/or suffix, eg. `security.server.crt_pass` would match `security.server.crt_pass_file`, too, due to using `indexOf()`.  It may not be a problem in this use case, but I think it should be addressed, since it is a generic method.

Reply:
Fixed